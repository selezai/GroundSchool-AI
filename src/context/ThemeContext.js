import React, { createContext, useContext } from 'react';

// Create the theme context
const ThemeContext = createContext();

// Theme provider component
export const ThemeProvider = ({ children }) => {
  // Theme values - always using dark blue aesthetic
  const theme = {
    // Always dark mode, no toggle needed
    isDarkMode: true,
    colors: {
      // Dark theme colors with original blue aesthetic
      background: '#0A0F24',
      card: 'rgba(255, 255, 255, 0.05)',
      text: '#FFFFFF',
      subText: '#E2E8F0',
      primary: '#00FFCC',
      border: 'rgba(255, 255, 255, 0.1)',
      accent: 'rgba(0, 255, 204, 0.2)',
      lightText: '#E2E8F0', // For secondary text
    },
  };

  return (
    <ThemeContext.Provider value={theme}>
      {typeof children === 'function' ? children(theme) : children}
    </ThemeContext.Provider>
  );
};

// Custom hook to use the theme context
export const useTheme = () => {
  const context = useContext(ThemeContext);
  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};
