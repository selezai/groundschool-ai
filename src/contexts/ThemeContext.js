import React, { createContext, useContext, useState } from 'react';

// Default theme colors
const defaultTheme = {
  colors: {
    primary: '#3498db',
    secondary: '#2ecc71',
    background: '#FFFFFF',
    card: '#F9FAFB',
    text: '#1F2937',
    border: '#E5E7EB',
    notification: '#FF9500',
    error: '#EF4444',
    success: '#10B981',
    warning: '#F59E0B',
    info: '#3B82F6',
  },
  spacing: {
    xs: 4,
    sm: 8,
    md: 16,
    lg: 24,
    xl: 32,
  },
  fontSizes: {
    xs: 12,
    sm: 14,
    md: 16,
    lg: 18,
    xl: 20,
    xxl: 24,
  },
  borderRadius: {
    sm: 4,
    md: 8,
    lg: 12,
    full: 9999,
  },
};

// Create the context
const ThemeContext = createContext(defaultTheme);

// Provider component
export const ThemeProvider = ({ children, initialTheme = defaultTheme }) => {
  const [theme, setTheme] = useState(initialTheme);

  const updateTheme = (newTheme) => {
    setTheme((prevTheme) => ({
      ...prevTheme,
      ...newTheme,
    }));
  };

  return (
    <ThemeContext.Provider value={{ ...theme, updateTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

// Custom hook for using the theme
export const useTheme = () => {
  const context = useContext(ThemeContext);
  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};

export default ThemeContext;
