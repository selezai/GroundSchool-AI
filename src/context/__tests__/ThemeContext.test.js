/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, renderHook } from '@testing-library/react';
import renderer, { act } from 'react-test-renderer';
import { ThemeProvider, useTheme } from '../ThemeContext';

describe('ThemeContext', () => {
  it('provides theme values to children', () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <ThemeProvider>
          <TestComponent />
        </ThemeProvider>
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  it('provides theme values to function children', () => {
    let receivedTheme;
    act(() => {
      renderer.create(
        <ThemeProvider>
          {(theme) => {
            receivedTheme = theme;
            return null;
          }}
        </ThemeProvider>
      );
    });

    expect(receivedTheme).toEqual({
      isDarkMode: true,
      colors: {
        background: '#0A0F24',
        card: 'rgba(255, 255, 255, 0.05)',
        text: '#FFFFFF',
        subText: '#E2E8F0',
        primary: '#00FFCC',
        border: 'rgba(255, 255, 255, 0.1)',
        accent: 'rgba(0, 255, 204, 0.2)',
        lightText: '#E2E8F0',
      },
    });
  });

  it('useTheme hook returns theme context', () => {
    const { result } = renderHook(() => useTheme(), { 
      wrapper: ({ children }) => <ThemeProvider>{children}</ThemeProvider>
    });

    expect(result.current).toEqual({
      isDarkMode: true,
      colors: {
        background: '#0A0F24',
        card: 'rgba(255, 255, 255, 0.05)',
        text: '#FFFFFF',
        subText: '#E2E8F0',
        primary: '#00FFCC',
        border: 'rgba(255, 255, 255, 0.1)',
        accent: 'rgba(0, 255, 204, 0.2)',
        lightText: '#E2E8F0',
      },
    });
  });

  it('useTheme hook throws error when used outside provider', () => {
    // In React 18 testing library, we need to check for the error differently
    expect(() => {
      renderHook(() => useTheme());
    }).toThrow('useTheme must be used within a ThemeProvider');
  });
});

// Test component that uses the theme
const TestComponent = () => {
  const theme = useTheme();
  return (
    <div style={{ backgroundColor: theme.colors.background }}>
      Test Component
    </div>
  );
};
