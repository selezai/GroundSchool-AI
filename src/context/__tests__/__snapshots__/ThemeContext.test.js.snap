// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeContext provides theme values to children 1`] = `null`;
